src/python/pants/vcs/git_test.py

```diff
@@ -374,3 +383,62 @@ async def worktree_id_string() -> str:
         worktree_id_2 = rule_runner.request(str, [])
 
         assert worktree_id_1 != worktree_id_2
+
+
+@pytest.mark.parametrize(
+    "diff,expected",
+    [
+        [
+            dedent(
+                """
+                --- left        2024-02-04 00:00:00.000000000 +0100
+                +++ right       2024-02-04 00:00:00.000000000 +0100
+                @@ -1,0 +2 @@
+                +two
+                """
+            ),
+            FrozenDict({"right": (Hunk(TextBlock.from_count(1, 0), TextBlock.from_count(2, 1)),)}),
+        ],
+        [
+            dedent(
+                """
+                --- left        2024-02-04 00:00:00.000000000 +0100
+                +++ right       2024-02-04 00:00:00.000000000 +0100
+                @@ -2 +1,0 @@
+                -two
+                """
+            ),
+            FrozenDict({"right": (Hunk(TextBlock.from_count(2, 1), TextBlock.from_count(1, 0)),)}),
+        ],
+        [
+            dedent(
+                """
+                --- left        2024-02-04 00:00:00.000000000 +0100
+                +++ right       2024-02-04 00:00:00.000000000 +0100
+                @@ -2 +2 @@
+                -two
+                +four
+                """
+            ),
+            FrozenDict({"right": (Hunk(TextBlock.from_count(2, 1), TextBlock.from_count(2, 1)),)}),
+        ],
+        [
+            dedent(
+                """
+                --- left        2024-02-04 00:00:00.000000000 +0100
+                +++ right       2024-02-04 00:00:00.000000000 +0100
+                @@ -2,2 +2,2 @@
+                -two
+                -three
+                +five
+                +six
+                """
+            ),
+            FrozenDict({"right": (Hunk(TextBlock.from_count(2, 2), TextBlock.from_count(2, 2)),)}),
+        ],
+    ],
```

Comment:
perhaps throw in a few cases with more interesting filenames, as well as multiple hunks per file, and multiple files in a diff.